server/ipqueue.go

```diff
@@ -80,6 +82,16 @@ func ipqLimitByLen[T any](max int) ipQueueOpt[T] {
 	}
 }
 
+// This option allows providing a function which is called to re-sort the
+// queued values each time something is pushed to the IPQ.
+// WARNING: This has the potential to be slow on very heavily utilized IPQs.
+// WARNING: It is only safe to use this with popOne() instead of pop().
```

Comment:
Not sure about this feature. In the use-case you have, there is still no guarantee that a `1sec` delay is not stuck behind a `3sec`, and by that I mean the currently popOne() may wait for 3sec. So if respecting the delay is really important, then the IPQueue approach, even with adding this sort feature, may not be great.

Also, you would move the "least delayed" at the front, but then that makes the others in the queue be delayed by that much. Say you have 3sec, 4sec in the queue, then add 1sec, 2sec, you now have 1sec, 2sec, 3sec, 4sec. You understand that each popOne() will sleep by that much and so the first that was going to be pop'ed and wait for 3sec is now itself delayed by 3sec before being pop'ed, etc...